lib/app/view/app.dart

```diff
@@ -0,0 +1,44 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+import 'package:flutter_localizations/flutter_localizations.dart';
+import 'package:qr_code/home/home.dart';
+import 'package:qr_code_repository/qr_code_repository.dart';
+
+class App extends StatelessWidget {
+  const App({
+    required this.qrCodeRepository,
+    super.key,
+  });
+
+  final QrCodeRepository qrCodeRepository;
+
+  @override
+  Widget build(BuildContext context) {
+    return RepositoryProvider.value(
+      value: qrCodeRepository,
+      child: const AppView(),
+    );
+  }
+}
+
+class AppView extends StatelessWidget {
+  const AppView({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return const MaterialApp(
```

Comment:
**question**: do you usally define routes in a separated folder? how do you do route management in your apps?

Reply:
Normally I use go_router because we can keep all routes more visible and reusable. But for this challange I just kept it simple.

Using go_router or routemaster should be the ideal solution because we can get the advantages of using Navigator 2.0 and make easy to use deep links for example :)